scripts/linters/custom_eslint_checks/rules/sorted-dependencies.js

```diff
@@ -0,0 +1,87 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Lint check to ensure that the dependencies
+ * are in sorted order.
+ */
+
+'use strict';
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: (
+        'Lint check to ensure that the dependencies are in sorted order'),
+      category: 'Possible Errors',
+      recommended: true,
+    },
+    fixable: null,
+    schema: [],
+    messages: {
+      unSortedDependencies: (
+        'Please ensure that the injected dependencies should be in the' +
+        ' following manner: dollar imports, local imports and' +
+        ' constant imports, all in sorted-order.')
+    },
+  },
+
+  create: function(context) {
+    var selector = (
+      'CallExpression[callee.property.name=/(controller|directive|factory)/]'
+    );
+    return {
+      [selector]: function(node) {
+        var args = node.arguments;
+        if (args.length !== 2 || args[1].type !== 'ArrayExpression') {
```

Comment:
Add a comment here explaining why you added this condition here.

Reply:
Done!

Reply:
Write something around this: `Checking if component function takes 2 arguments and the second argument is an ArrayExpression` and then add a short example.

Reply:
@AdityaDubey0 In my previous comment I suggested you to rephrase the comment but I gave the wrong example. As you can see, that example is saying 'what' this 'if' statement is doing, instead we should write comments that shows 'why' we are doing something. The if statement is clear here but why we are doing such a check is not clear. Hence add a comment so that it explains why we are adding this 'if' statement.

Reply:
Done!

Reply:
Add a short example, in the comment.

Reply:
Had an offline discussion with @AdityaDubey0. No need to add a code example here.

Reply:
Thanks